cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java

```diff
@@ -125,19 +131,115 @@ private static void buildAngleRefs(Bus bus, Map<String, String> angleRefs, Cgmes
+            return new TopologicalIsland(topologicalNode, Collections.singletonList(topologicalNode), context);
+        }
+
+        void addNode(String topologicalNode, Bus bus, boolean updateLoadFlowStatus) {
+            topologicalNodes.add(topologicalNode);
+            if (updateLoadFlowStatus) {
+                updateLoadFlowStatus(bus);
+            }
+        }
+
+        void updateLoadFlowStatus(Bus bus) {
+            if (loadFlowStatus.equals(DIVERGED)) {
+                return;
+            }
+            if (!(isValidVoltage(bus.getV()) && isValidAngle(bus.getAngle()) && isInAccordanceWithKirchoffsFirstLaw(bus))) {
+                loadFlowStatus = DIVERGED;
+            }
+        }
+
+        boolean isValidVoltage(double v) {
+            return v >= 0.01;
+        }
+
+        boolean isValidAngle(double a) {
+            return Double.isFinite(a);
+        }
+
+        boolean isInAccordanceWithKirchoffsFirstLaw(Bus bus) {
+            // Instead of checking switch flows, we check that the corresponding bus view bus is balanced
+            // Maybe we will check the same bus view bus more than once
+            Optional<Bus> bus0 = getBusViewBus(bus);
+            if (bus0.isEmpty()) {
+                LOG.warn("Can not check if bus is in accordance with Kirchoff's first law. No BusView bus can be found for: {}", bus);
+                return false;
+            }
+            Complex sumTerminalFlows = bus0.get().getConnectedTerminalStream()
+                    .filter(this::isValidTerminal)
+                    .map(t -> new Complex(t.getP(), t.getQ()))
+                    .reduce(Complex.ZERO, Complex::add);
+            boolean isInAccordance = Math.abs(sumTerminalFlows.getReal()) < maxPMismatchConverged && Math.abs(sumTerminalFlows.getImaginary()) < maxQMismatchConverged;
+            if (!isInAccordance && LOG.isInfoEnabled()) {
+                LOG.info("Bus {} is not in accordance with Kirchoff's first law. Mismatch = {}", bus, sumTerminalFlows);
+                bus0.get().getConnectedTerminalStream()
+                        .filter(this::isValidTerminal)
+                        .forEach(t -> LOG.info(String.format("  %7.2f  %7.2f  %s %s %s", t.getP(), t.getQ(), t.getConnectable().getType(), t.getConnectable().getNameOrId(), t.getConnectable().getId())));
+                LOG.info(String.format("  %7.2f  %7.2f", sumTerminalFlows.getReal(), sumTerminalFlows.getImaginary()));
+            }
+            return isInAccordance;
+        }
+
+        Optional<Bus> getBusViewBus(Bus bus) {
+            if (bus.getVoltageLevel().getTopologyKind().equals(TopologyKind.BUS_BREAKER)) {
+                return Optional.of(bus.getVoltageLevel().getBusView().getMergedBus(bus.getId()));
+            } else {
+                return bus.getConnectedTerminalStream().map(t -> t.getBusView().getBus()).filter(Objects::nonNull).findFirst();
+            }
+        }
+
+        boolean isValidTerminal(Terminal t) {
```

Comment:
We can leave that...

Reply:
Excepted for bus bar section. If we have an issue with static var compensator, the issue is in the power flow engine. Or in a more clever treatment of NaN values.

Reply:
separate p and q sums, changed by sum over non-NaN values and check not all terminals have NaN value.